src/fuzzer.rs

```diff
@@ -3,43 +3,79 @@ use {
 	colored::*,
 	futures::{StreamExt, stream},
 	std::sync::Arc,
+	url::Url
 };
 
-use crate::parser;
+use crate::{parser, payload};
 
-static CHECK_SCRIPT: &str = "(window.ppfuzz || Object.prototype.ppfuzz) == 'reserved' && true || false";
+const CHECK_SCRIPT: &str = "(window.ppfuzz || Object.prototype.ppfuzz) == 'reserved' && true || false";
+const FINGERPRINT: &str = include_str!("fingerprint.js");
 
 pub async fn new(urls: Vec<String>, browser: Browser, opt: parser::Options) {
 	let browser = Arc::new(browser);
+	let mut targets: Vec<String> = vec![];
 	let mut stream = stream::iter(urls.into_iter()
 		.map(|url| (url, Arc::clone(&browser)))
 		.map(|(url, browser)| async move {
 			let is_err: bool = false;
+			let mut gadgets: Vec<String> = vec![];
 			let mut detail: String = String::new();
 			let page = match browser.new_page(&url).await {
 				Ok(res) => res,
 				Err(err) => {
 					detail = err.to_string();
-					return Ok((url, false, !is_err, detail))
+					return Ok((url, false, !is_err, detail, gadgets))
 				}
 			};
 			let vuln: bool = match page.evaluate(CHECK_SCRIPT).await {
-				Ok(res) => {
-					page.close().await.unwrap();
-					res.into_value().unwrap()
-				},
+				Ok(res) => res.into_value().unwrap(),
 				Err(_) => false
 			};
 
+			if vuln {
+				gadgets = match page
+					.evaluate(FINGERPRINT)
+					.await {
+						Ok(res) => {
+							page.close().await.unwrap();
+							res.into_value().unwrap()
+						},
+						Err(_) => gadgets
+					};
+			}
+
 			Ok::<_, Box<dyn std::error::Error>>
-			((url, vuln, is_err, detail))
+			((url, vuln, is_err, detail, gadgets))
 		}
 	)).buffer_unordered(opt.concurrency);
 
 	while let Some(res) = stream.next().await {
-		if let Ok((ref url, vuln, is_err, detail)) = res {
+		if let Ok((ref url, vuln, is_err, detail, gadgets)) = res {
 			if vuln {
-				println!("[{}] {}", "VULN".green(), url)
+				println!("[{}] {}", "VULN".green(), url);
+
+				let payloads = payload::get();
+				let mut target = Url::parse(url).unwrap();
+				let mut new_target = target.clone();
+				let mut queries: Vec<(_, _)> = target
+					.query_pairs()
+					.collect();
+
+				for p in payloads {
+					queries = queries.into_iter()
+						.filter(|q| q.0 != p)
+						.collect();
+				}
```

Comment:
this is fine, but I think you probably can also use something like `target.query_pairs().into_iter().filter(|q| payloads.iter().all(|p|q.0 !=p) )`